apps/src/code-studio/components/progress/ScriptOverviewHeader.jsx

```diff
@@ -84,19 +81,24 @@ class ScriptOverviewHeader extends Component {
     showCourseUnitVersionWarning: PropTypes.bool,
     showScriptVersionWarning: PropTypes.bool,
     showRedirectWarning: PropTypes.bool,
-    versions: PropTypes.arrayOf(scriptVersionShape).isRequired,
+    versions: PropTypes.arrayOf(assignmentVersionShape).isRequired,
     showHiddenUnitWarning: PropTypes.bool,
-    courseName: PropTypes.string
+    courseName: PropTypes.string,
+    locale: PropTypes.string
   };
 
   componentDidMount() {
     $('#lesson-heading-extras').appendTo(ReactDOM.findDOMNode(this.protected));
   }
 
-  onChangeVersion = event => {
-    const scriptName = event.target.value;
-    if (scriptName !== this.props.scriptName) {
-      window.location.href = `/s/${scriptName}`;
+  onChangeVersion = versionYear => {
+    const script = this.props.versions.find(v => v.year === versionYear);
+    if (
+      script &&
+      script.name.length > 0 &&
+      script.name !== this.props.scriptName
+    ) {
+      window.location.href = `/s/${script.name}`;
```

Comment:
this looks like it will lose the `section_id` and the `user_id`. here's one example of where we take extra steps to preserve the section_id: https://github.com/code-dot-org/code-dot-org/blob/65ad7252b01b0765fabef975310d073b5fdecdfe/apps/src/templates/teacherDashboard/OwnedSectionsTable.jsx#L83-L84 the same would need to be done for user_id.